pkg/ndm-grpc/server/services/listBlockDevice.go

```diff
@@ -0,0 +1,281 @@
+/*
+Copyright 2019 The OpenEBS Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"strings"
+
+	"github.com/openebs/node-disk-manager/pkg/apis/openebs/v1alpha1"
+	protos "github.com/openebs/node-disk-manager/pkg/ndm-grpc/protos/ndm"
+	"k8s.io/klog"
+
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/openebs/node-disk-manager/cmd/ndm_daemonset/controller"
+
+	"github.com/openebs/node-disk-manager/pkg/hierarchy"
+	"github.com/openebs/node-disk-manager/pkg/ndm-grpc/server"
+)
+
+// Node helps in using types defined in Server
+type Node struct {
+	server.Node
+}
+
+// NewNode is a constructor
+func NewNode() *Node {
+	return &Node{server.Node{}}
+}
+
+type disks []protos.BlockDevice
+
+//AllBlockDevices contains all the relationships and device types
+type AllBlockDevices struct {
+	Parents    []string
+	Partitions []string
+	LVMs       []string
+	RAIDs      []string
```

Comment:
LVM, RAID, Loop and Sparse are not relationship between devices. Its a deviceType.

Reply:
The struct is supposed to have all the relationships and device types. Please suggest a better name for it.